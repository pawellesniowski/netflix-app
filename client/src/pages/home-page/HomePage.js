import React from "react";
import {connect} from 'react-redux';
import SearchPanel from "../../components/SearchPanel/SearchPanel";
import MoviesGrid from "../../components/MoviesGrid/MoviesGrid";
import Footer from "../../components/Footer/Footer";
import SortBar from "../../components/SortBar/SortBar";
import {fetchMovies, SortMovies } from './homePageActions';

class HomePage extends React.Component {
    componentDidMount() {
        const searchDetails = this.props.location.search;
        if(searchDetails) {
            this.props.fetchMovies(searchDetails);
        }
    }

    componentDidUpdate(prevProps) {
        const searchDetails = this.props.location.search;
        if(prevProps.location.search !== searchDetails) {
            this.props.fetchMovies(searchDetails)
        }
    }

    render() {
        const { movies, SortMovies, sortedBy } = this.props;
        const submitFilmSearch = ({searchBy, searchTerm}) => {
            this.props.history.push(`/search/movies?search=${searchTerm}&searchBy=${searchBy}`);
        };
        return (
            <React.Fragment>
                <SearchPanel onSubmit={submitFilmSearch}/>
                <SortBar moviesNumber={movies.length} sorting={SortMovies} sortedBy={sortedBy}/>
                <MoviesGrid movies={movies}/>
                <Footer/>
            </React.Fragment>
        );
    }
}

const mapStateToProps = (state) => {
    return {
        movies: state.homePage.movies,
        sortedBy: state.homePage.sortedBy,
    }
};

const mapDispatchToProps = {
    fetchMovies,
    SortMovies,
};

export default connect(mapStateToProps, mapDispatchToProps)(HomePage);
