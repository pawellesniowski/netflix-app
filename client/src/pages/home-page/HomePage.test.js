import React from 'react';
import {shallow} from "enzyme/build";
import HomePage from "./HomePage";


describe('HomePage snapshot', () => {
    it("Renders HomePage Component", () => {
        const component = shallow(<HomePage />);
        expect(component).toMatchSnapshot();
    });
});
// HomePage content tests done in cypress integration.
