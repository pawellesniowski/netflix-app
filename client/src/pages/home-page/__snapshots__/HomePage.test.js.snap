// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage snapshot Renders HomePage Component 1`] = `ShallowWrapper {}`;
